cdap-common/src/main/java/io/cdap/cdap/common/NamespaceRepositoryNotFoundException.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.common;
+
+import io.cdap.cdap.proto.id.NamespaceId;
+
+/**
+ * Thrown when a namespace repository configuration is not found in CDAP.
+ */
+public class NamespaceRepositoryNotFoundException extends NotFoundException  {
+
+  private final NamespaceId namespace;
+
+  public NamespaceRepositoryNotFoundException(NamespaceId id, String reason) {
+    super(id, reason);
```

Comment:
The `reason` should be defined inside this class instead of passing from the parameter.